documentation/src/main/wiki/CoreArchitecture.md

```diff
@@ -0,0 +1,9 @@
+[WORKING IN PROGRESS]
+#Knot.x Core Architecture
+
+Diagram below depicts Knot.x core modules. Knot.x by default comes with Core and Core Adapters modules.
+Every module can be disabled / replaced with a simple JSON configuration change.
+
+[[assets/knotx-architecture-full.png|alt=Knot.x Core Architecture]]
```

Comment:
We are adding some png images. They are probably created with some diagrams editor and exported to png format.

Could we attach original format for the diagrams editor in order to update those images easily?